sound/soc/sof/virtio-fe.c

```diff
@@ -0,0 +1,734 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2017-2019 Intel Corporation. All rights reserved.
+ *
+ * Author:	Libin Yang <libin.yang@intel.com>
+ *		Luo Xionghu <xionghu.luo@intel.com>
+ *		Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ *		Guennadi Liakhovetski <guennadi.liakhovetski@linux.intel.com>
+ */
+
+/*
+ * virt IO FE driver
+ *
+ * The SOF driver thinks this driver is another audio DSP, however the calls
+ * made by the SOF driver core do not directly go to HW, but over a virtIO
```

Comment:
@lyakh I could use some education about what you really mean by "The SOF driver thinks this driver is another audio DSP". As far as I understand, the SOF driver only knows about one audio DSP. How does it handled a second DSP?

Reply:
The intention here is that the mailbox and doorbell are virtualised, but the upper layers dont know this. They just send IPCs as if there were being sent to a physical doorbell and mailbox.

Reply:
agree with Ranjani, the sentence makes no sense to me. I think the intent is something like

'the SOF driver relies on an IPC protocol, but the IPC commands are not transmitted using hardware mailboxes/doorbell registers but Virtio command queues'

Reply:
@ranj063 @plbossart the wording can be improved, sure. An important part there is "to the virtIO BE driver," i.e. even though the degree of the SOF host driver involvement in the IPC traffic from the guest to the DSP is minimal, some IPC messages are intercepted by the host and not sent to the DSP, so, it isn't strictly speaking a perfect pass-through. But when the message is indeed forwarded to the DSP, that is a matter of a simple copy between the VirtQ and the mailbox.